src/components/utils/page-visited-app.tsx

```diff
@@ -0,0 +1,8 @@
+import { withPageVisited as withPageVisitedOriginal, PageVisitedProps } from "./page-visited";
+
+export function withPageVisited<T extends PageVisitedProps>(
+    Component: React.ComponentType<T>,
+    key: string
+) {
+    return withPageVisitedOriginal(Component, "vaccination-app", key);
```

Comment:
> add a change listener to the router-history to keep track in the dataStore of the unvisited pages

I hadn't thought of it, it looks like a very centralized way to do it! Not sure if there are some drawbacks, it seems we would need to perform an "inverse routing" then (route to component). But definitely, something to check the next time we need something like this. Let's make a mental note for all the routing thing, I think we all agree we don't like hardcoded paths all over the app (instead of named paths in one place)